src/main/java/org/mockito/internal/MockitoCore.java

```diff
@@ -62,6 +63,11 @@ public boolean isTypeMockable(Class<?> typeToMock) {
         MockCreationSettings<T> creationSettings = impl.build(typeToMock);
         T mock = createMock(creationSettings);
         mockingProgress().mockingStarted(mock, creationSettings);
+        
+        if(typeToMock.isAnnotation()) {
+            configureMockWithAnnotationDefaults(typeToMock, mock);
+        }
```

Comment:
This works well but I'm not sure if this is the best position to hook in this functionality - I'm happy to adjust if there is a better position to perform this.